src/move_insertion.jl

```diff
@@ -93,36 +87,41 @@ function cost_move_insertion(solution::Solution, position::Int,
 
         #---------------------------------------------------------- Cost of insertion
 
-    # objective >= 1 #Must improve or keep HPRC
-
-    #What about insertion at b0 (first place available)
-    delta2_for_first[b0] = compute_delta2_for_b0(
-        solution, instance, # instance
-        M1, sequence, position, # calcul
-        1, instance.nb_HPRC # for options
-    )
-    delta1_for_first[b0] = compute_delta1(
-        solution, instance, # instance
-        M1, sequence, b0, position, # calcul
-        1, instance.nb_HPRC # for options
-    )
-
-    # for all other valid positions
-    for index in (b0+1):(instance.nb_cars)
-        # Delta 1 is compute as for b0
-        delta1_for_first[index] = compute_delta1(solution, instance,
-                                                 M1, sequence, index, position,
-                                                 1, instance.nb_HPRC)
-    end
-    for index in (b0+1):(instance.nb_cars)
-        delta2_for_first = compute_delta2(solution, instance,
-                                          M1, sequence, index, position,
-                                          1, instance.nb_HPRC,
-                                          delta2_for_first)
+    # Cost on HPRC
+    if objectives[1]
+        #What about insertion at b0 (first place available)
+        delta2_for_first[b0] = compute_delta2_for_b0(
+            solution, instance, # instance
+            M1, sequence, position, # calcul
+            1, instance.nb_HPRC # for options
+        )
+        delta1_for_first[b0] = compute_delta1(
+            solution, instance, # instance
+            M1, sequence, b0, position, # calcul
+            1, instance.nb_HPRC # for options
+        )
+
+        # for all other valid positions
+        for index in (b0+1):(instance.nb_cars)
+            # Delta 1 is compute as for b0
+            delta1_for_first[index] = compute_delta1(solution, instance,
+                                                     M1, sequence, index, position,
+                                                     1, instance.nb_HPRC)
+        end
+        for index in (b0+1):(instance.nb_cars)
+            delta2_for_first = compute_delta2(solution, instance,
+                                              M1, sequence, index, position,
+                                              1, instance.nb_HPRC,
+                                              delta2_for_first)
+        end
+        # The cost is variation of deletion + delta1 (new sequence) + delta2 (modified sequences)
+        for i in b0:(instance.nb_cars)
+            cost_on_objective[i, 1] = delta1_for_first[i] + delta2_for_first[i] + violations_caused_on_first
+        end
```

Comment:
This can be put later when `delta*_for_second` are computed too (if they are).

```Julia
if objectives[1]
    if objectives[2]
        for i in b0:(instance.nb_cars)
            cost_on_objective[i, 1] = delta1_for_first[i] + delta2_for_first[i] + violations_caused_on_first
            cost_on_objective[i, 2] = delta1_for_second[i] + delta2_for_second[i] + violations_caused_on_second
        end
    else
        for i in b0:(instance.nb_cars)
            cost_on_objective[i, 1] = delta1_for_first[i] + delta2_for_first[i] + violations_caused_on_first
        end
    end
elseif objectives[2]
    for i in b0:(instance.nb_cars)
        cost_on_objective[i, 2] = delta1_for_second[i] + delta2_for_second[i] + violations_caused_on_second
    end
end
```

Reply:
I think it will just more written code without any optimization.
```
if objectives[1]
    if objectives[2]
        do 1 and 2
    else
        do 1
    end
elseif objectives[2]
    do 2
end
```
Far all cases we do exactly the same two tests. The following code is less redundant:
```
if objectives[1]
    do 1
end
if objectives[2]
    do 2
end
```

However, I approve placement after `delta*_for_second` computation.